#This is my code to get nucleotide diversity estimates from GA and GAR cad86c Sanger data
#4/18/2019 MF
#alignment files generated by CLUSTAL Omega (1.2.4) multiple sequence alignment available 
#from EMBL-EBI

library(ape) #version 5.0
library(pegas)
library(FastaUtils)
source("https://bioconductor.org/biocLite.R")
biocLite("Biostrings")


#starting with primers 1b
CAD_GAR <- read.dna(file = "PATH/TO/FILE.aln", 
                      format = "clustal")

CAD_GA <- read.dna(file = "PATH/TO/FILE.aln", 
                      format = "clustal")

#Counting the number of segregating sites in these two datasets
seg_res <- seg.sites(CAD_GAR)
length(seg_res)

seg_sus <- seg.sites(CAD_GA)
length(seg_sus)

#getting theta
theta_res <- theta.s(CAD_GAR)
theta_res

theta_sus <- theta.s(CADsus_GA)
theta_sus

#getting pi
nuc_res <- nuc.div(CAD_GAR)
nuc_res

nuc_sus <- nuc.div(CAD_GA)
nuc_sus

real_diff <- nuc_sus-nuc_res


#resampling fasta files to generate new nuc div estimates

#resampled (without replacement) nuc divs from permuted pops.

resamp_nucdiv_diff <-numeric(0)

for (i in 1:1000){
fasta.sample(infile = "/PATH_TO/ALL.fasta", nseq = 92,
             replacement = FALSE, file.out = "out.fasta")
GA_resamp <- head(read.FASTA(file = "~/out.fasta"), n = 48)
GAR_resamp <- tail(read.FASTA(file = "~/out.fasta"), n = 44)
diff <- (nuc.div(GA_resamp))-(nuc.div(GAR_resamp))
resamp_nucdiv_diff <- c(nucdiv_diff, diff)
}


hist(resamp_nucdiv_diff, main = "CAD86c Distribution of NucDiv Diff")

sig <- sum(resamp_diff_nucdiv > real_diff)# shows the p-value for the permutation test.
