#This is my code to get nucleotide diversity estimates from GA and GAR cad86c Sanger data
#4/18/2019 MF
#alignment files generated by CLUSTAL Omega (1.2.4) multiple sequence alignment available 
#from EMBL-EBI

library(ape) #version 5.0
library(pegas)


#starting with primers 1b
CAD_GAR <- read.dna(file = "PATH/TO/FILE.aln", 
                      format = "clustal")

CAD_GA <- read.dna(file = "PATH/TO/FILE.aln", 
                      format = "clustal")

#Counting the number of segregating sites in these two datasets
seg_res <- seg.sites(CAD_GAR)
length(seg_res)

seg_sus <- seg.sites(CAD_GA)
length(seg_sus)

#getting theta
theta_res <- theta.s(CAD_GAR)
theta_res

theta_sus <- theta.s(CADsus_GA)
theta_sus

#getting pi
nuc_res <- nuc.div(CAD_GAR)
nuc_res

nuc_sus <- nuc.div(CAD_GA)
nuc_sus
